tracer/src/Datadog.Tracer.Native/debugger_rejit_preprocessor.cpp

```diff
@@ -238,14 +238,14 @@ const std::unique_ptr<RejitHandlerModuleMethod>
 DebuggerRejitPreprocessor::CreateMethod(const mdMethodDef methodDef, RejitHandlerModule* module,
                                         const FunctionInfo& functionInfo, const MethodProbeDefinition& methodProbe)
 {
-    return std::make_unique<DebuggerRejitHandlerModuleMethod>(methodDef, module, functionInfo);
+    return std::make_unique<DebuggerRejitHandlerModuleMethod>(methodDef, module, functionInfo, std::make_unique<DebuggerMethodRewriter>(m_corProfiler));
```

Comment:
Do we have any numbers of allocations? we moved from a singleton to creating a `DebuggerMethodRewriter` instance on each call of this method. Hopefully it's called just once.

Reply:
Well we already make an allocation for the `DebuggerRejitHandlerModuleMethod` so I don't expect the extra allocation to have an impact. That said, I guess we could store the instance in an additional field in the rejit preprocessor.